tests/product/prestoadmin_installer.py

```diff
@@ -82,69 +82,35 @@ def _build_installer_in_docker(self, cluster, online_installer=None,
         if isinstance(cluster, ConfigurableCluster):
             online_installer = True
 
-        container_name = 'installer'
-        cluster_type = 'installer_builder'
-        bare_image_provider = NoHadoopBareImageProvider("build")
-
-        installer_container, created_bare = DockerCluster.start_cluster(
-            bare_image_provider, cluster_type, 'installer', [])
-
-        if created_bare:
-            installer_container.commit_images(
-                bare_image_provider, cluster_type)
+        temp_dir = os.path.join(tempfile.mkdtemp(), "presto-admin")
 
         try:
             shutil.copytree(
                 prestoadmin.main_dir,
-                os.path.join(
-                    installer_container.get_local_mount_dir(container_name),
-                    'presto-admin'),
+                temp_dir,
                 ignore=shutil.ignore_patterns('tmp', '.git', 'presto*.rpm')
             )
 
-            # Pin pip to 7.1.2 because 8.0.0 removed support for distutils
-            # installed projects, of which the system setuptools is one on our
-            # Docker image. pip 8.0.1 or 8.0.2 replaced the error with a
-            # deprecation warning, and also warns that Python 2.6 is
-            # deprecated. While we still need to support Python 2.6, we'll pin
-            # pip to a 7.x version, but we should revisit this once we no
-            # longer need to support 2.6:
-            # https://github.com/pypa/pip/issues/3384
-            installer_container.run_script_on_host(
-                'set -e\n'
-                # use explicit versions of dependent packages
-                'pip install --upgrade pycparser==2.18 cffi==1.11.5\n'
-                'pip install --upgrade pycparser==2.18 PyNaCl==1.2.1\n'
-                'pip install --upgrade pycparser==2.18 idna==2.1 cryptography==2.1.1\n'
-                'pip install --upgrade pip==9.0.2\n'
-                'pip install --upgrade wheel==0.23.0\n'
-                'pip install --upgrade setuptools==20.1.1\n'
-                'mv %s/presto-admin ~/\n'
-                'cd ~/presto-admin\n'
-                'make %s\n'
-                'cp dist/prestoadmin-*.tar.gz %s'
-                % (installer_container.mount_dir,
-                   'dist' if online_installer else 'dist-offline',
```

Comment:
so we never ran tests in `online_installer` here which looks like some dead code https://github.com/prestosql/presto-admin/pull/22/files#diff-0e9360ea0e4c1e02b85edc7c23fa0984R82 ? I dont want to clean all that up, just asking

Reply:
The online installer was definitely exercised. Take a look at `test_online_installer.py`, which calls this method directly with `online_installer=True`. However, I don't know if the particular IF statement you link to ever gets exercised.